fiftyone/core/odm/mixins.py

```diff
@@ -497,6 +498,9 @@ def _rename_fields(cls, sample_collection, paths, new_paths):
         dataset_doc.app_config._rename_paths(paths, new_paths)
         dataset_doc.save()
 
+        if schema_paths:
+            cls._rename_indexes(paths, new_paths)
```

Comment:
Index updates are performed at the very end so that, if the dataset is big and the user ctrl + c's the operation sometime while the index operations are being performed, the data model changes will at least be finished.